tools/clang/test/CodeGenSPIRV/texture.get-sample-position.hlsl

```diff
@@ -2,8 +2,111 @@
+// CHECK:   [[pos4_3:%\d+]] = OpConstantComposite %v2float %float_0_125 %float_0_375
+// CHECK:     [[pos4:%\d+]] = OpConstantComposite %_arr_v2float_uint_4 [[pos4_0]] [[pos4_1]] [[pos4_2]] [[pos4_3]]
+
+// CHECK:   [[pos8_0:%\d+]] = OpConstantComposite %v2float %float_0_0625 %float_n0_1875
+// CHECK:   [[pos8_1:%\d+]] = OpConstantComposite %v2float %float_n0_0625 %float_0_1875
+// CHECK:   [[pos8_2:%\d+]] = OpConstantComposite %v2float %float_0_3125 %float_0_0625
+// CHECK:   [[pos8_3:%\d+]] = OpConstantComposite %v2float %float_n0_1875 %float_n0_3125
+// CHECK:   [[pos8_4:%\d+]] = OpConstantComposite %v2float %float_n0_3125 %float_0_3125
+// CHECK:   [[pos8_5:%\d+]] = OpConstantComposite %v2float %float_n0_4375 %float_n0_0625
+// CHECK:   [[pos8_6:%\d+]] = OpConstantComposite %v2float %float_0_1875 %float_0_4375
+// CHECK:   [[pos8_7:%\d+]] = OpConstantComposite %v2float %float_0_4375 %float_n0_4375
+// CHECK:     [[pos8:%\d+]] = OpConstantComposite %_arr_v2float_uint_8 [[pos8_0]] [[pos8_1]] [[pos8_2]] [[pos8_3]] [[pos8_4]] [[pos8_5]] [[pos8_6]] [[pos8_7]]
+
+// CHECK: [[pos16_00:%\d+]] = OpConstantComposite %v2float %float_0_0625 %float_0_0625
+// CHECK: [[pos16_01:%\d+]] = OpConstantComposite %v2float %float_n0_0625 %float_n0_1875
+// CHECK: [[pos16_02:%\d+]] = OpConstantComposite %v2float %float_n0_1875 %float_0_125
+// CHECK: [[pos16_03:%\d+]] = OpConstantComposite %v2float %float_0_25 %float_n0_0625
+// CHECK: [[pos16_04:%\d+]] = OpConstantComposite %v2float %float_n0_3125 %float_n0_125
+// CHECK: [[pos16_05:%\d+]] = OpConstantComposite %v2float %float_0_125 %float_0_3125
+// CHECK: [[pos16_06:%\d+]] = OpConstantComposite %v2float %float_0_3125 %float_0_1875
+// CHECK: [[pos16_07:%\d+]] = OpConstantComposite %v2float %float_0_1875 %float_n0_3125
+// CHECK: [[pos16_08:%\d+]] = OpConstantComposite %v2float %float_n0_125 %float_0_375
+// CHECK: [[pos16_09:%\d+]] = OpConstantComposite %v2float %float_0 %float_n0_4375
+// CHECK: [[pos16_10:%\d+]] = OpConstantComposite %v2float %float_n0_25 %float_n0_375
+// CHECK: [[pos16_11:%\d+]] = OpConstantComposite %v2float %float_n0_375 %float_0_25
+// CHECK: [[pos16_12:%\d+]] = OpConstantComposite %v2float %float_n0_5 %float_0
+// CHECK: [[pos16_13:%\d+]] = OpConstantComposite %v2float %float_0_4375 %float_n0_25
+// CHECK: [[pos16_14:%\d+]] = OpConstantComposite %v2float %float_0_375 %float_0_4375
+// CHECK: [[pos16_15:%\d+]] = OpConstantComposite %v2float %float_n0_4375 %float_n0_5
+// CHECK:    [[pos16:%\d+]] = OpConstantComposite %_arr_v2float_uint_16 [[pos16_00]] [[pos16_01]] [[pos16_02]] [[pos16_03]] [[pos16_04]] [[pos16_05]] [[pos16_06]] [[pos16_07]] [[pos16_08]] [[pos16_09]] [[pos16_10]] [[pos16_11]] [[pos16_12]] [[pos16_13]] [[pos16_14]] [[pos16_15]]
+
+// CHECK:     [[zero:%\d+]] = OpConstantComposite %v2float %float_0 %float_0
+
+void main(int index : INDEX) {
+// CHECK:  %var_GetSamplePosition_data_2 = OpVariable %_ptr_Function__arr_v2float_uint_2 Function [[pos2]]
```

Comment:
Should we avoid variable initialization like this?

Reply:
meaning, replace with:
```spirv
%var... = OpVariable %_ptr_Function__arr_v2float_uint_2
OpStore %var [[pos2]]
```

Reply:
So there are two issues regarding inline initializer:

* Driver support
* spirv-opt support

For the first one, inline initializer for `Function` variables should be fine since we have CTS coverage for it for a long time:

https://github.com/KhronosGroup/VK-GL-CTS/blob/7763f6e68b28413787c638fe5c9f1b20a5579636/external/vulkancts/modules/vulkan/spirv_assembly/vktSpvAsmInstructionTests.cpp#L5441

For the second one, spirv-opt is already fixed to support inline initializers for all cases.

Reply:
But for `Private` variables, it is different. No CTS coverage results in buggy driver support.

Reply:
Hmm, anyway, I think using `OpStore` is still safer. Changed.

Reply:
I see. SGTM. Thanks